subprojects/gradle/sentry-config/src/main/kotlin/com/avito/android/sentry/ProjectExtensions.kt

```diff
@@ -7,47 +7,66 @@ import com.avito.kotlin.dsl.ProjectProperty
 import com.avito.kotlin.dsl.PropertyScope.ROOT_PROJECT
 import com.avito.kotlin.dsl.getBooleanProperty
 import com.avito.kotlin.dsl.getMandatoryStringProperty
-import com.avito.kotlin.dsl.lazy
+import com.avito.kotlin.dsl.getOptionalStringProperty
 import com.avito.kotlin.dsl.lazyProperty
 import io.sentry.SentryClient
+import okhttp3.HttpUrl
 import org.gradle.api.Project
 import org.gradle.api.provider.Provider
 
 fun Project.environmentInfo(): Provider<EnvironmentInfo> = lazyProperty("ENVIRONMENT_INFO_PROVIDER") { project ->
-    project.providers.lazy {
+    project.providers.provider {
         val git = Git.Impl(project.rootDir, project.logger::info)
         EnvironmentInfoImpl(project, git)
     }
 }
 
 val Project.sentry: Provider<SentryClient> by ProjectProperty.lazy(scope = ROOT_PROJECT) { project ->
-    project.providers.lazy {
-        val config = from(project)
-        sentryClient(config)
-    }
+    project.sentryConfig.map { sentryClient(it) }
 }
 
 val Project.sentryConfig: Provider<SentryConfig> by ProjectProperty.lazy(scope = ROOT_PROJECT) { project ->
-    project.providers.lazy {
-        from(project)
-    }
+    project.providers.provider { from(project) }
 }
 
 private fun from(project: Project): SentryConfig {
     return if (project.getBooleanProperty("avito.sentry.enabled")) {
         val info = project.environmentInfo().get()
         val tags = mutableMapOf<String, String>()
         tags["ide"] = info.isInvokedFromIde().toString()
-        info.teamcityBuildId()?.also { id ->
-            tags["build_id"] = id
+
+        val buildId = info.teamcityBuildId()
+
+        val buildIdTag = "build_id"
+
+        if (!buildId.isNullOrBlank()) {
+            tags[buildIdTag] = buildId
         }
-        SentryConfig.Enabled(
+
+        val config = SentryConfig.Enabled(
             dsn = project.getMandatoryStringProperty("avito.sentry.dsn"),
             environment = info.environment.publicName,
             serverName = info.node ?: "unknown",
             release = info.commit ?: "unknown",
             tags = tags
         )
+
+        val projectUrl = project.getOptionalStringProperty("avito.sentry.projectUrl")
+
+        if (!projectUrl.isNullOrBlank() && !buildId.isNullOrBlank()) {
+            project.gradle.buildFinished {
+
+                val url = HttpUrl.parse(projectUrl)
+                    ?.newBuilder()
+                    ?.addQueryParameter("query", "$buildIdTag:$buildId")
+                    ?.build()
+                    ?.toString()
+
+                project.logger.lifecycle("Build errors: $url")
```

Comment:
This url is incorrect in local builds: `build_id:0`

I see two options:
- Surrogate build id for local builds
- Filter by available attributes: host, user, ...

What do you think?

Reply:
Lets create local id from user-machine-and-date-time

Reply:
MBS-8612